### PR TITLE
Allow using the Twig use statement when extending legacy templates

### LINKED ISSUE #6447: The Twig use statement does not work when extending legacy templates
### Affected version(s)

5.2

### Description

When extending from a legacy template horizontal reuse seems to be broken.

Reproducer:
```twig
{# templates/fe_page.html.twig #}
{% extends "@Contao/fe_page.html5" %}
{% use "@Contao/component/_figure.html.twig" %}
```

This will lead to an error saying `figure` was not set, which it isn't, but the component block's function should not even get called at that point. :thinking:

### PR DESCRIPTION
**Pull request by m-vo**
Fixes #6447 

IMHO that's the right fix - I want to manually test it some more, though.

### CODE REVIEW

File: core-bundle/tests/Twig/Interop/PhpTemplateProxyNodeTest.php [hunk 1]
```diff
@@ -33,7 +33,7 @@ public function testCompilesProxyCode(): void
                     function(callable $block) use ($context): string {
                         if ($this->env->isDebug()) { ob_start(); } else { ob_start(static function () { return ''; }); }
                         try { $block($context); return ob_get_contents(); } finally { ob_end_clean(); }
-                    }, $blocks
+                    }, array_filter($blocks, static fn(callable $block): bool => method_exists(self::class, $block[1]))
```
Comment (ausi) [inline core-bundle/tests/Twig/Interop/PhpTemplateProxyNodeTest.php]: Matching via the method name seems a bit strange, can’t we just directly check for the block keys that get passed to the legacy template?

```suggestion
                    }, array_intersect_key($blocks, $this->blocks)
```

This had the same affect in my tests.

File: core-bundle/src/Twig/Interop/PhpTemplateProxyNode.php [hunk 2]
```diff
@@ -51,7 +51,7 @@ public function compile(Compiler $compiler): void
             ->write('if ($this->env->isDebug()) { ob_start(); } else { ob_start(static function () { return \'\'; }); }'."\n")
             ->write('try { $block($context); return ob_get_contents(); } finally { ob_end_clean(); }'."\n")
             ->outdent()
-            ->write('}, $blocks'."\n")
+            ->write('}, array_filter($blocks, static fn(callable $block): bool => method_exists(self::class, $block[1]))'."\n")
```
Comment (leofeyer) [inline core-bundle/src/Twig/Interop/PhpTemplateProxyNode.php]:
```suggestion
            ->write('}, array_intersect_key($blocks, $this->blocks)'."\n")
```

File: core-bundle/src/Twig/Interop/PhpTemplateProxyNode.php [hunk 3]
```diff
@@ -35,7 +35,7 @@ public function compile(Compiler $compiler): void
         //         function(callable $block) use ($context): string {
         //             if ($this->env->isDebug()) { ob_start(); } else { ob_start(static function () { return ''; }); }
         //             try { $block($context); return ob_get_contents(); } finally { ob_end_clean(); }
-        //         }, $blocks
+        //         }, array_filter($blocks, static fn(callable $block): bool => method_exists(self::class, $block[1]))
```
Comment (leofeyer) [inline core-bundle/src/Twig/Interop/PhpTemplateProxyNode.php]:
```suggestion
        //         }, array_intersect_key($blocks, $this->blocks)
```

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @m-vo.

File: core-bundle/tests/Twig/Interop/PhpTemplateProxyNodeTest.php (hunk 1 above)
Comment (m-vo) [reply core-bundle/tests/Twig/Interop/PhpTemplateProxyNodeTest.php]: That was my first attempt as well. Although, I don't recall the reason anymore, I think there were cases where this did not work. :see_no_evil: But let's find out…

**Pull request merged**